src/editor/actions/fetch-cell-parser.js

```diff
@@ -49,7 +46,61 @@ export function emptyLine(line) {
   return line.replace(/\s+/g, "") === "";
 }
 
-export function parseFetchCellLine(line) {
+export function missingFetchType(line) {
+  return !line.trim().match(/^\w+\s*: /);
+}
+
+export function validFetchType(line) {
+  const fetchType = line.trim().split(": ")[0];
+  return fetchType.trimLeft().match(/^(css|js|arrayBuffer|blob|json|text)$/);
+}
+
+export function validVariableName(line) {
+  const fetchType = line
+    .trim()
+    .split(": ")[0]
+    .trim();
+  const fetchCommand = line
+    .trim()
+    .split(": ")
+    .slice(1)
+    .join(": ");
+  if (fetchType.match(/^(css|js)$/)) {
+    return true;
+  }
+  const varName = fetchCommand.substring(0, fetchCommand.indexOf("=")).trim();
+  return isValidIdentifier(varName);
+}
+
+export function validFetchUrl(line) {
+  /*
+    Assume fetch type and variable name are valid,
+    For script fetch (css & js), valid fetch content is:
+      "<fetch-url>"
+    For data fetch (arrayBuffer, blob, json & text), valid fetch content is:
+      "<variable-name> = <fetch-url>"
+  */
+  const fetchType = line.split(": ")[0];
+  const fetchContent = line.replace(new RegExp(`^${fetchType}: `), "").trim();
+
+  let fetchUrl;
+  if (fetchType.match(/^(js|css)$/)) {
```

Comment:
Similar to the above, could we specify an array called `URL_FETCH_TYPES` and do:

```suggestion
  if (URL_FETCH_TYPES.includes(fetchType)) {
```